usr/src/uts/common/brand/lx/syscall/lx_prctl.c

```diff
@@ -288,6 +290,54 @@ lx_prctl(int opt, uintptr_t data)
 		return (0);
 	}
 
+	case LX_PR_GET_NO_NEW_PRIVS: {
+		/*
+		 * Some Linux applications (such as Elasticsearch) use
+		 * PR_SET_NO_NEW_PRIVS to enable the NoNewPrivs flag, and then
+		 * query the flag status, and fail unless the query indicates
+		 * the flag is enabled. Since we return success in
+		 * LX_PR_SET_NO_NEW_PRIVS, we assume here that the application
+		 * has intended to enable the flag, so we return 1 indicating
+		 * that the flag is enabled.
+		 */
+		return (1);
+	}
+
+	case LX_PR_SET_SECCOMP: {
+		if (arg2 == SECCOMP_MODE_FILTER) {
+			if (arg3 == (uintptr_t)NULL) {
+				/*
+				 * prctl(2) says PR_SET_SECCOMP should indicate
+				 * EFAULT if arg3 is an invalid address.
+				 */
+				return (set_errno(EFAULT));
+			}
+
+			/*
+			 * Some Linux applications (such as Elasticsearch)
+			 * attempt to install seccomp BPF rules. Since Illumos
+			 * doesn't have an analogous system call rule engine,
+			 * for now we just pretend that the rule installation
+			 * succeeded.
+			 */
```

Comment:
Perhaps a dtrace probe here?

        DTRACE_PROBE(lx__SECCOMP);  /* So we can track who needs this */

Mention also in the above paragraph that elasticsearch uses this to prevent fork/exec in practice.

Reply:
Also mention the word "lie" in this paragraph for "git grep -w lie" searching.  "we just pretend and lie that the rule..."